yext/chat-headless
# Analytics support

- expose `report` function to send event payload through `@yext/analytics` to Analytics API
- getNextMessage and streamNextMessage will send a CHAT_RESPONSE analytics event
- CHAT_IMPRESSION is short circuit such that the event is only allow to send once

J=CLIP-300
TEST=manual

from test site:
- see that sending a message (standard and stream) also send an analytics event with the correct payload
- see that sending multiple CHAT_IMPRESSION event through `report` function only allow for one event to be sent as shown through network tab

**postponing unit tests to later this week**

![Screenshot 2023-06-27 at 6 17 09 PM](https://github.com/yext/chat-headless/assets/36055303/56bc7178-83fd-4812-9037-cc144a1841b2)
![Screenshot 2023-06-27 at 6 17 30 PM](https://github.com/yext/chat-headless/assets/36055303/849c9571-cba5-474d-934c-ffbc20424b4a)
